cloud/ovirt/ovirt_clusters.py

```diff
@@ -0,0 +1,207 @@
+            ) if self._module.params['datacenter'] else None,
+            management_network=otypes.Network(
+                name=self._module.params['network'],
+            ) if self._module.params['network'] else None,
+            cpu=otypes.Cpu(
+                architecture=self._module.params['cpu_arch'],
+                type=self._module.params['cpu_type'],
+            ) if (
+                self._module.params['cpu_arch'] or self._module.params['cpu_type']
+            ) else None,
+            version=otypes.Version(
+                major=self.__get_major(self._module.params['compatibility_version']),
+                minor=self.__get_minor(self._module.params['compatibility_version']),
+            ) if self._module.params['compatibility_version'] else None,
+            switch_type=otypes.SwitchType(
+                self._module.params['switch_type']
+            ) if self._module.params['switch_type'] else None,
+        )
+
+    def update_check(self, entity):
+        return (
+            equal(self._module.params.get('comment'), entity.comment) and
+            equal(self._module.params.get('description'), entity.description) and
+            equal(self._module.params.get('switch_type'), str(entity.switch_type)) and
+            equal(self._module.params.get('cpu_arch'), str(entity.cpu.architecture)) and
+            equal(self._module.params.get('cpu_type'), entity.cpu.type) and
+            equal(self.__get_minor(self._module.params.get('compatibility_version')), self.__get_minor(entity.version)) and
+            equal(self.__get_major(self._module.params.get('compatibility_version')), self.__get_major(entity.version))
+        )
+
+
+def main():
+    argument_spec = ovirt_full_argument_spec(
+        state=dict(
+            choices=['present', 'absent'],
+            default='present',
+        ),
+        name=dict(default=None, required=True),
+        datacenter=dict(default=None),
+        description=dict(default=None),
+        comment=dict(default=None),
+        network=dict(default=None),
+        cpu_arch=dict(default=None),
+        cpu_type=dict(default=None),
+        switch_type=dict(default=None),
+        compatibility_version=dict(default=None),
+    )
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+    )
+    check_sdk(module)
+    check_params(module)
+
+    try:
+        connection = create_connection(module.params.pop('auth'))
+        clusters_service = connection.system_service().clusters_service()
+        clusters_module = ClustersModule(
+            connection=connection,
+            module=module,
+            service=clusters_service,
+        )
+
+        state = module.params['state']
+        if state == 'present':
+            ret = clusters_module.create()
+        elif state == 'absent':
+            ret = clusters_module.remove()
+
+        module.exit_json(**ret)
+    except Exception as e:
```

Comment:
catchall exceptions are normally frowned upon as the error message here might not reflect the action attempted, we prefer friendlier error messages on each possible failure action that help the user figure out what to fix in order for the task to work.